src/hooks/useFunctions.tsx

```diff
@@ -181,14 +181,9 @@ export const FunctionsProvider = ({children, componentUri}: FunctionProviderProp
             functionService.findAllFunctions()
                 .then(func => {
                     _setAllFunctions(func)
-                    _setFunctionsAndComponents([])
-                    func.forEach(f => {
-                        functionService.getComponent(f.iri)
-                            .then(comp => {
-                                _setFunctionsAndComponents(value => [...value, [f,comp]])})
-                            .catch(() => {
-                                _setFunctionsAndComponents(value => [...value, [f,null]])})
-                    })
+                    let funcsAndComs : [Function,Component][] = []
+                    func.forEach( f => funcsAndComs.push([f, f['http://onto.fel.cvut.cz/ontologies/fta-fmea-application/hasComponent']]))
```

Comment:
There's a problem with this because our Function model does not contain component, so we get undefined component everytime. There are two options: 

- remake model and implement it like FailureMode (more work, better performance)
- let the old code (slow)

Reply:
I suggest the first option.

Reply:
First option seams better for me too. 
I suggest to merge this request into main because it fixes the unresponsiveness problem and create a new issue to change the function model.

Reply:
Regarding the function model changes. If I understand correctly changes in the model are required only on the client side as as the model for failure mode and function on the server side are derived from model of behavior. 
Can we do the same on the client side?

Reply:
Agree. 

Yes, changes are required only on the client side. 
"Can we do the same on the client side?" I will take a look at this, but it should be possible